### DISCUSSION TITLE
How to hide 'not matched item' in the result  list.

### DISCUSSION THREAD

**Discussion by iplaywind** [tomik23/Leaflet.Autocomplete#28] [Announcements]
Thank you for providing the excellent search functionality. I have a question that requires your assistance.
Please visit the webpage [http://route.capas.oome.net/sharecode/Leaflet.Autocomplete-master/docs/index.html](http://route.capas.oome.net/sharecode/Leaflet.Autocomplete-master/docs/index.html). I have used the code from this repository and run the docs/index.html file, with only modifications script.js file onlines 8-14 of the , changing the data source from api to the built-in search.json file. 
When I search for "gmina," all 10 results are listed in the list, with 3 of them not matched "gmina".
 Since my own geojson file contains thousands of data entries, I do not wish to display all the items. Instead, I would like to display only the entries that match the search keyword, as shown in the image. 
![Snipaste_2024-04-17_21-23-39](https://github.com/tomickigrzegorz/Leaflet.Autocomplete/assets/6620570/375e0da3-cc58-4fac-a497-706addeb16df)
How should I modify the JavaScript code ?

**Comment by tomickigrzegorz** [comment]
Take a look at this library because it's the same thing - [autocomplete](https://tomickigrzegorz.github.io/autocomplete/). There are detailed examples of how to search in json files.

**Comment by tomickigrzegorz** [comment]
The code below should help:
```js
return new Promise((resolve) => {
      fetch(api)
        .then((response) => response.json())
        .then((data) => {
          const results = data.features.filter((element) => {
            return element.properties.display_name
              .toLowerCase()
              .includes(currentValue);
          });

          resolve(results);
        })
        .catch((error) => {
          console.error(error);
        });
```
these are the basics of searching in a file :)
